app/models/goodcity_setting.rb

```diff
@@ -0,0 +1,4 @@
+class GoodcitySetting < ActiveRecord::Base
+  validates :key, presence: true
+  validates :key, uniqueness: true
```

Comment:
```suggestion
  validates :key, uniqueness: true, presence: true
```

You can batch validation params together as a hash